scripts/ioctool

```diff
@@ -26,46 +38,154 @@ function iocdir {
+   # Look for a complete match, anchored on both sides.
+   # Look for a match anchored at the end.
+   # If there is more than one, return "ambiguous"
+   #
+   # Make a temp file with just the PVs and camera names that match.
+   tmp=/tmp/cv.$$
+   cat $VIEWERCFG | grep -i "$CAMNAME" | grep -v '#' | awk -F, '{ if (split($2,a,";")==1) n=$2; else n=a[2]; gsub(" ","",n); gsub(" ","",$4); printf "%s %s\n", n, $4; }' >$tmp
+   c=$(wc -l <$tmp)
+   if [ "$c" -eq 0 ]; then 
+      VIEWERCFG=/reg/g/pcds/pyps/config/*/camviewer.cfg
+      cat $VIEWERCFG | grep -i "$CAMNAME" | grep -v '#' | awk -F, '{ if (split($2,a,";")==1) n=$2; else n=a[2]; gsub(" ","",n); gsub(" ","",$4); printf "%s %s\n", n, $4; }' >$tmp
+      c=$(wc -l <$tmp)
+      if [ "$c" -eq 0 ]; then
+         echo "No match for $2"
+         rm -f $tmp
+         exit
+      fi    
+   fi
+   if [ "$c" -eq 1 ]; then
+      campv=$(awk '{print $1;}' <$tmp)
+      rm -f $tmp
+      return
+   fi
+   # OK, ambiguous.  Look for a complete match on the second word.
+   if [ "$(grep -E -i "$CAMNAME\$" < $tmp | wc -l)" -eq 1 ]; then
+      campv=$(grep -E -i "$CAMNAME\$" < $tmp | awk '{print $1;}')
+      rm -f $tmp
+      return
+   fi
+   # What about just a match at the end? (That is, if we are searching
+   # for "yag1", prefer "mec_yag1" to "mec_yag11".)
+   if [ "$(grep -E -i "$CAMNAME\$" < $tmp | wc -l)" -eq 1 ]; then
+      campv=$(grep -E -i "$CAMNAME\$" < $tmp | awk '{print $1;}')
+      rm -f $tmp
+      return
+   fi
+   echo '"'"$2"'" is ambiguous:'
+   rm -f $tmp
+   exit
+}
+
 usage(){
 cat << EOF
-usage: ${BASH_SOURCE[0]} <ioc>|<pv> [option]
+usage: $0 <ioc>|<pv>|<camera name> [option]
 
-Script that returns information about an ioc given its name or a PV it hosts
+Script that returns information about an ioc given its name, PV, or camera name
 
 default option is 'name', list of options:
-name   : returns the name of the ioc
-dir    : returns the directory the ioc is running from
-cddir  : open the directory the ioc is running from (start with "source" before calling script with this option)
-cfg    : returns the file name of the ioc .cfg (or st.cmd)
-less   : opens the ioc .cfg (or st.cmd) in less
-data   : returns the path of the appropriate iocData directory if it exists
-telnet : starts a telnet session with the ioc 
+name     : returns the name of the ioc
+dir      : returns the directory the ioc is running from
+cddir    : open the directory the ioc is running from (start with "source" before calling script with this option)
+cfg      : returns the file name of the ioc .cfg (or st.cmd)
+less     : opens the ioc .cfg (or st.cmd) in less
+cat      : opens the ioc .cfg (or st.cmd) in cat
+data     : returns the path of the appropriate iocData directory if it exists
+autosave : opens the most recent autosave file in less
+archive  : opens the most recent archive file in less
+log      : opens the most recent log file in less
+telnet   : starts a telnet session with the ioc 
```

Comment:
The new options here should be added to the `ioctool` section of the repo readme